src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -125,7 +125,13 @@ def _merge_extension_data(
         all_unit_ids = new_sorting_analyzer.unit_ids
         not_new_ids = all_unit_ids[~np.isin(all_unit_ids, new_unit_ids)]
 
+        # this creates a new metrics dictionary, but the dtype for everything will be
+        # object
         metrics = pd.DataFrame(index=all_unit_ids, columns=old_metrics.columns)
+        # we can iterate through the columns and convert them back to numbers with
+        # pandas.to_numeric. coerce allows us to keep the nan values.
+        for column in metrics.columns:
+            metrics[column] = pd.to_numeric(metrics[column], errors="coerce")
```

Comment:
```suggestion
            metrics[column] = pd.to_numeric(metrics[column], errors="coerce")
            if np.all(np.mod(metrics[column], 1) == 0):
                metrics[column] = metrics[column].astype(int)
```
@zm711 @JoeZiminski what about this to restore integers?

Reply:
This actually doesn't work because at this point everything is a nan until data is loaded.